File: Automation/ols.R
library(fixest) # fixest is the go to for estimation in R
library(haven) # Read .dta

df <- read_dta("https://github.com/scunning1975/mixtape/raw/master/card.dta")

(no_cov <- feols(lwage ~ educ, data = df, vcov = "HC1"))
(cov <- feols(lwage ~ educ + exper + black + south + married + smsa, data = df, vcov = "HC1"))


# Regression Table
fixest::etable(no_cov, cov,
			   title = "OLS estimates of effect of college on log wages")

## Latex option
fixest::etable(no_cov, cov,
			   title = "OLS estimates of effect of college on log wages", tex = T)

## File
fixest::etable(no_cov, cov,
			   title = "OLS estimates of effect of college on log wages",
			   tex = T, file = "estout/education.tex"
)

File: Automation/ols.py
import pandas as pd
import statsmodels.api as sm
from stargazer.stargazer import Stargazer
​
# Load data
card = pd.read_stata('https://raw.github.com/scunning1975/mixtape/master/card.dta')
​
# Models
no_cov = sm.OLS.from_formula('lwage ~ + educ', data = card).fit(cov_type = "HC3")
cov = sm.OLS.from_formula('lwage ~ educ + exper + black + south + married + smsa', data = card).fit(cov_type = "HC3")
​
# Create LaTeX table
stargazer = Stargazer([no_cov, cov])  
stargazer.title('OLS estimates of effect of college on log wages')
stargazer.custom_columns(['No covariates', 'Covariates'], [1, 1])
stargazer.add_custom_notes(["Data is from NLS.  Heteroskedastic standard errors shown in parenthesis."])
stargazer.dependent_variable_name('Dependent variable: log(wage)')
# Export table as .tex file
with open('estout/education.tex', 'w') as f:
    f.write(stargazer.render_latex())
